backend/zgw/zaken-api/src/main/kotlin/com/ritense/zakenapi/web/rest/ZaakDocumentResource.kt

```
@@ -40,26 +51,84 @@ class ZaakDocumentResource(
+    ): ResponseEntity<InputStreamResource> {
+
+        val documentInputStream = zaakDocumentService.downloadInformatieObject(pluginConfigurationId, caseDocumentId, documentId)
+        val documentMetadata = zaakDocumentService.getInformatieObject(pluginConfigurationId,caseDocumentId, documentId)
+
+        val responseHeaders = HttpHeaders()
+        val contentDisposition = ContentDisposition.attachment().filename(documentMetadata.bestandsnaam).build()
+        responseHeaders.contentDisposition = contentDisposition
+
+        val documentMediaType = try {
+            MediaType.valueOf(URLConnection.guessContentTypeFromName(documentMetadata.bestandsnaam))
+        } catch (exception: RuntimeException) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)